lib/screens/home/home_bloc.dart

```diff
@@ -78,7 +92,56 @@ class HomePreferences {
   HomePreferences({
     @required this.sortType,
     @required this.filterType,
-  })
-      : assert(sortType != null, "sortType cannot be null."),
+  })  : assert(sortType != null, "sortType cannot be null."),
+        assert(filterType != null, "filterType cannot be null.");
+}
+
+/// Used as a data class for specifying details about queries in the PageClass
+class PageQuery {
+  final SortType sortType;
+  final FilterType filterType;
+  final int pageNumber;
+
+  PageQuery({
+    @required this.sortType,
+    @required this.filterType,
+    @required this.pageNumber,
+  })  : assert(sortType != null, "sortType cannot be null."),
         assert(filterType != null, "filterType cannot be null.");
+
+  @override
+  bool operator ==(Object other) =>
+      identical(this, other) ||
+      other is PageQuery &&
+          runtimeType == other.runtimeType &&
+          sortType == other.sortType &&
+          filterType == other.filterType &&
+          pageNumber == other.pageNumber;
+
+  @override
+  int get hashCode =>
+      sortType.hashCode ^ filterType.hashCode ^ pageNumber.hashCode;
+}
+
+class PageCache<PageQuery, Page> extends Cache {}
```

Comment:
based on the optimizations you can make to the bloc, I don't think this cache is even necessary.